### ISSUE TITLE
transforms[key_value_parser]: when values contains separator character(s) ,value occur data loss

### ISSUE THREAD

**Issue by zhoujw8792** [vectordotdev/vector#5347]
### Vector Configuration File

```toml
[sources.syslog_514]
  type = "syslog"
  address = "0.0.0.0:514"
  mode = "udp"

[transforms.source_ip]
  type = "split"
  inputs = ["syslog_514"]
  drop_field = false
  field = "source_ip"
  field_names = ["dev_ip"]
  separator = ":"

[transforms.kv_parser_transform]
  type = "key_value_parser"
  inputs = ["source_ip"]
  drop_field = false
  field = "message"
  field_split = "="
  #separator = " "
  #target_field = "data"
  #trim_key = "\""
  trim_value = "\""

[sinks.syslog_514_out]
  inputs   = ["kv_parser_transform"]
  type     = "console"
  encoding = "json"
```

### Debug Sink  Output

{
    "Action": "PERMIT",
    **"Content": "Session",**
    "DstIP": "10.10.254.7",
    "DstPort": "137",
    **"GenTime": "2019-08-19"**,
    "PolicyID": "1",
    "Protocol": "UDP",
    "SerialNum": "100018002000001906146520",
    "SrcIP": "10.10.254.2",
    "SrcPort": "137",
    "appname": "FILTER",
    "facility": "local5",
    "host": "QSFW-1",
    "hostname": "QSFW-1",
    "message": "SerialNum=100018002000001906146520 **GenTime=\"2019-08-19 15:17:44\"** SrcIP=10.10.254.2 DstIP=10.10.254.7 Protocol=UDP SrcPort=137 DstPort=137 PolicyID=1 Action=PERMIT **Content=\"Session Backout\"**",
    "severity": "alert",
    "source_ip": "10.65.60.180",
    "source_type": "syslog",
    "timestamp": "2020-08-19T07:17:44Z"
}

### Expected Behavior
json data key_value
   "GenTime":"2019-10-24 14:25:03"
    "Content": "Session Backout"

### Actual Behavior
json data key_value
   "GenTime":"2019-10-24"
   "Content": "Session"

### Example Data

<318>Oct 24 14:25:03 QSFW-1 FILTER: SerialNum=100018002000001906146520 GenTime="2019-10-24 14:25:03" SrcIP=10.10.254.2 DstIP=10.10.254.7 Protocol=UDP SrcPort=137 DstPort=137 PolicyID=3 Action=PERMIT Content="Session Backout"
<710>Aug 19 16:04:52 QSFW-1 APPCONTROL: SerialNum=100018002000001906146520 GenTime="2019-08-19 16:04:52" SrcIP=10.10.254.2 DstIP=10.10.254.7 Protocol=UDP SrcPort=137 DstPort=137 InInterface=ge0/2 AAA=any App=netbios-ns Action=PERMIT PolicyID=1 AppAction=any Content="The packet is allowed"

### Additional Context

 Vector running in windows and centos 7.3

download package 
   vector-0.11.0-x86_64-pc-windows-msvc
   vector-0.11.0-x86_64-unknown-linux-musl

**Comment by binarylogic** [comment]
@FungusHumungus can you make sure that the new `parse_key_value` function in #5368 does not suffer from the same issue? And close it with the work there. We will be deprecating the `key_value_parser` transform.

**Comment by FungusHumungus** [comment]
This is going to be fixed by #5760.

It should be noted that this PR introduces a Remap function to parse the key_value line correctly. The `key_value_parser` transform will be deprecated in favour of Remap. To fix the issue, replace the above transform with 

```
[transforms.kv_parser_transform]
  type = "remap"
  inputs = ["source_ip"]
  source = '''
     merge(., parse_key_value(.message))
  '''
```

A Remap transform is not limited to doing one thing however. It is likely that the split tranform can also be removed in favour of the `split` Remap function, which can also be incorporated into this transform.
